cilium/CFP-33716-distributed-packet-tracing.md

```diff
@@ -0,0 +1,424 @@
+# CFP-2024-06-12: Distributed Packet Tracing
+
+## Meta
+**SIG:**  
+**Sharing:** Public  
+**Begin Design Discussion:** Jun 12, 2024  
+**End Design Discussion:**  
+**Cilium Release:**  
+**Authors:** Ben Bigdelle <bigdelle@google.com>, Mark St. John <markstjohn@google.com> based on the work of Nathan Perkins <nathanperkins@google.com>  
+**Reviewers:**  
+
+## Summary
+We aim to enhance cross-cluster and multi-system observability for flows that contain an `ip_trace_id` IP Option (referred to as a tagged flow), thus supporting distributed tracing across networks. These tagged flows will trigger adding the ID value to Flow events messages and client (Hubble / Cilium Monitor) representation. This allows correlating flow events across systems which integrate with the span observability.
+
+Additionally, within Cilium, these flows will generate datapath debugging events regardless of the global debug-verbose setting. This will improve debuggability of production systems.
+
+## Motivation
+There are currently limitations for gathering and correlating flow events:
+- NAT transformations and DSR can all change the 4-tuple, so that a single Hubble or tcpdump filter cannot be used to trace the packet along the entire path – even when using a system such as Hubble Relay.
+- Enabling verbose datapath debugging is not possible in production environments due to traffic volume which leads to dropping flows.
+
+## Goals
+- For tagged packets, add the ID to emitted flow events.
+- For tagged packets, emit datapath debugging events regardless of the debug-verbose setting.
+- Support both IPv4 and IPv6.
+- Add hubble filter for trace ID.
+- Allow users to define which flow IDs they want to monitor for events.
+
+## Non-goals
+- Automatically apply IP option to packets (see Future Milestones).
```

Comment:
So when I read the title it mentions _distributed_ packet tracing, but this really just seems like "regular" packet tracing, since there's to trace ID propagation. Generally when I think of "distributed" tracing, I think of having observability beyond the initial source and destination. This doesn't tell you anything beyond the connection between the initial source and destination. It might be worth mention that somewhere, or rephrasing the title of the CFP to reflect this.

Reply:
Agreed, makes sense to drop the term distributed as this is only a part of what could be integrated into a larger system tracing system (e.g. with fabric integration). 

Although nested virtualized nodes (e.g. via kubevirt) may also capture flows in the path and then it becomes a bit greyer 😛

Reply:
This has been fixed and will be closed as resolved.